### PR TITLE
Increase parity between dev and prod instances when triggering sync/link

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [x] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

This PR
- Deprecates and prepares the ground for the removal of `__clerk_referrer_primary` which ended up to be redundant
- Triggers sync/link for development instances only if client uat is missing
- Refactors the steps we take in order to load clerk-js properly for each flow

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/core/clerk.ts [hunk 1]
```diff
@@ -1160,48 +1153,78 @@ export default class Clerk implements ClerkInterface {
...
+    if (!this.proxyUrl && !this.domain) {
+      clerkMissingProxyUrlAndDomain();
+    }
+  };
+
   #loadInStandardBrowser = async (): Promise<boolean> => {
-    // Dev Browser handling
+    /**
+     * 1. Create the devBrowser.
+     * At this point the devBrowser is not yet setup, but its API is ready for use
```
Comment (SokratisVidros) [inline packages/clerk-js/src/core/clerk.ts:1183]: @panteliselef let's explain why this step is required in the comment. If not mistake the devBrowserHandler is used by some multi-domain helpers. This should be stated explicitly.

File: packages/clerk-js/src/core/clerk.ts [hunk 2]
```diff
@@ -1160,48 +1153,78 @@ export default class Clerk implements ClerkInterface {
...
+    if (this.#shouldSyncWithPrimary()) {
+      await this.#syncWithPrimary();
+      // ClerkJS is not considered loaded during the sync/link process with the primary domain
+      return false;
+    }
+
+    /**
+     * 3. Setup dev browser.
+     * This is not needed for production instances hence the .clear()
+     * At this point we have already attempted to pre-populate devBrowser with a fresh JWT, if Step 3 was successful this will not be overwritten
```
Comment (SokratisVidros) [inline packages/clerk-js/src/core/clerk.ts:1205]: Please also elaborate on this by stating that in a satellite app we don't want to get the JWT from FAPI but from URL Based Session Syncing as we are trying to sync the primary and satellite dev browser values.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

**Review by nikosdouvlis — COMMENTED** [review]
Good work @panteliselef 
I'll do another pass as this requires more time, but this makes more sense now

File: packages/clerk-js/src/core/clerk.ts [hunk 3]
```diff
@@ -1112,40 +1122,23 @@ export default class Clerk implements ClerkInterface {
   };
 
   #shouldSyncWithPrimary = (): boolean => {
+    // TODO: Remove this in the minor release
```
[the remark this answers is not in the document]
Comment (panteliselef) [reply packages/clerk-js/src/core/clerk.ts:1125]: We need this PR to go out in a prod release, everyone to use that new version, and then remove the code marked with this comments. Does that sound ok ?

[the remark this answers is not in the document]
Comment (nikosdouvlis) [reply packages/clerk-js/src/core/clerk.ts:1125]: How are we keeping track of who used this at least once? Do we need to?

[the remark this answers is not in the document]
Comment (panteliselef) [reply packages/clerk-js/src/core/clerk.ts:1125]: I don't think we need to. 

Everything marked as deprecated there will only affect development instances. So no worries about breaking clients.
I believe with the new way we deliver clerk-js once we release a version everyone will get it quickly.

And regarding to the changes made in the backend package, we will not mind if the will be using the old version as long as the run the latest clerk-js (this flow is driven by clerk-js)

**Review by SokratisVidros — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**